pulumi-eks-go/main.go

```diff
@@ -0,0 +1,439 @@
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: IAM Role
+		// Purpose: An IAM role is an IAM identity that you can create in your account that has specific permissions.
+		// Docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+
+		// IAM Role for EKS
+		eksRole, err := iam.NewRole(ctx, prefix+"eks-iam-eksRole", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2008-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "eks.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+		})
+		if err != nil {
+			return err
+		}
+		eksPolicies := []string{
+			"arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+			"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+		}
+		for i, eksPolicy := range eksPolicies {
+			_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf(prefix+"rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+				PolicyArn: pulumi.String(eksPolicy),
+				Role:      eksRole.Name,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		// Resource: Security Groups
+		// Purpose: A security group acts as a virtual firewall, controlling the traffic that is allowed to reach and leave the resources that it is associated with.
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html
+
+		// Create a Security Group that we can use to actually connect to our cluster
+		clusterSg, err := ec2.NewSecurityGroup(ctx, prefix+"-cluster-sg", &ec2.SecurityGroupArgs{
+			VpcId: vpc.ID(),
+			Egress: ec2.SecurityGroupEgressArray{
+				ec2.SecurityGroupEgressArgs{
+					Protocol:   pulumi.String("-1"),
+					FromPort:   pulumi.Int(0),
+					ToPort:     pulumi.Int(0),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+				},
+			},
+			Ingress: ec2.SecurityGroupIngressArray{
+				ec2.SecurityGroupIngressArgs{
+					Protocol:   pulumi.String("tcp"),
+					FromPort:   pulumi.Int(80),
+					ToPort:     pulumi.Int(80),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+				},
+			},
+			Tags: pulumi.StringMap{
+				"Name": pulumi.String(prefix + "-cluster-sg"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: EKS Cluster
+		// Purpose: The Amazon EKS control plane & Amazon EKS nodes that are registered with the control plane
+		// Docs: https://docs.aws.amazon.com/eks/latest/userguide/clusters.html
+
+		// EKS cluster
+		var subnetIds pulumi.StringArray
+
+		cluster, err := eks.NewCluster(ctx, prefix+"-cluster", &eks.ClusterArgs{
+			RoleArn: pulumi.StringOutput(eksRole.Arn),
+			VpcConfig: &eks.ClusterVpcConfigArgs{
+				PublicAccessCidrs: pulumi.StringArray{
+					pulumi.String("0.0.0.0/0"),
+				},
+				SecurityGroupIds: pulumi.StringArray{
+					clusterSg.ID().ToStringOutput(),
+				},
+				SubnetIds: pulumi.StringArray(
+					append(
+						subnetIds,
+						privSubnet1.ID(),
```

Comment:
This is my workaround to pass a slice - empty subnetIds. May be better to have a helper that takes the subnet IDs and returns an obj I can pass into `pulumi.StringArray()`